subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultArtifactTransformDependencies.java

```diff
@@ -35,14 +37,16 @@
 public class DefaultArtifactTransformDependencies implements ArtifactTransformDependencies {
     private final ComponentArtifactIdentifier artifactId;
     private final ResolvableDependencies resolvableDependencies;
+    private final ImmutableAttributes attributes;
 
-    public DefaultArtifactTransformDependencies(ComponentArtifactIdentifier artifactId, ResolvableDependencies resolvableDependencies) {
+    public DefaultArtifactTransformDependencies(ComponentArtifactIdentifier artifactId, ResolvableDependencies resolvableDependencies, ImmutableAttributes attributes) {
         this.artifactId = artifactId;
         this.resolvableDependencies = resolvableDependencies;
+        this.attributes = attributes;
     }
 
     @Override
-    public Iterable<File> getDependencies() {
+    public Iterable<File> getFiles() {
```

Comment:
Now that we have a `ArtifactTransformDependenciesProvider` I think we should make this type eager/immutable. We must calculate the final list of files before executing the transform in any case, and we will need to iterate them at least twice (once when snapshotting inputs, and at least once when user code processes them).